packages/flutter/lib/src/material/pickers/date_range_picker_dialog.dart

```diff
@@ -0,0 +1,678 @@
+                        ),
+                        Flexible(
+                          child: Text(
+                            endDateText,
+                            style: endDateStyle,
+                            maxLines: 1,
+                            overflow: TextOverflow.ellipsis,
+                          ),
+                        ),
+                      ],
+                    ),
+                    const SizedBox(height: 16),
+                  ],
+                ),
+              ),
+            ),
+            if (orientation == Orientation.portrait)
+              Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 8.0),
+                child: entryModeIcon,
+              ),
+          ]),
+          preferredSize: const Size(double.infinity, 64),
+        ),
+      ),
+      body: CalendarDateRangePicker(
+        initialStartDate: selectedStartDate,
+        initialEndDate: selectedEndDate,
+        firstDate: firstDate,
+        lastDate: lastDate,
+        currentDate: currentDate,
+        onStartDateChanged: onStartDateChanged,
+        onEndDateChanged: onEndDateChanged,
+      ),
+    );
+  }
+}
+
+class _InputDateRangePickerDialog extends StatelessWidget {
+  const _InputDateRangePickerDialog({
+    Key key,
+    @required this.selectedStartDate,
+    @required this.selectedEndDate,
+    @required this.picker,
+    @required this.onConfirm,
+    @required this.onCancel,
+    @required this.onToggleEntryMode,
+    @required this.confirmText,
+    @required this.cancelText,
+    @required this.helpText,
+  }) : super(key: key);
+
+  final DateTime selectedStartDate;
+  final DateTime selectedEndDate;
+  final Widget picker;
+  final VoidCallback onConfirm;
+  final VoidCallback onCancel;
+  final VoidCallback onToggleEntryMode;
+  final String confirmText;
+  final String cancelText;
+  final String helpText;
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final ColorScheme colorScheme = theme.colorScheme;
+    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+    final Orientation orientation = MediaQuery.of(context).orientation;
+    final TextTheme textTheme = theme.textTheme;
+    final TextDirection textDirection = Directionality.of(context);
+
+    final Color dateColor = colorScheme.brightness == Brightness.light
+      ? colorScheme.onPrimary
+      : colorScheme.onSurface;
+    final TextStyle dateStyle = orientation == Orientation.landscape
+      ? textTheme.headline5?.apply(color: dateColor)
+      : textTheme.headline4?.apply(color: dateColor);
+    final String startDateText = utils.formatRangeStartDate(localizations, selectedStartDate, selectedEndDate);
+    final String endDateText = utils.formatRangeEndDate(localizations, selectedStartDate, selectedEndDate, DateTime.now());
+    final String dateText = (selectedStartDate == null || selectedEndDate == null)
```

Comment:
This inline conditional chain is getting a little out of hand... Maybe use a conditional block or two here?